bloom.go

```diff
@@ -0,0 +1,89 @@
+// Package bloom implements a Bloom filter data structure.
+package bloom
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"errors"
+	"log"
+	"math/bits"
+)
+
+// Filter represents a Bloom filter. Filter satisfies the encoding.BinaryMarshaler and BinaryUnmarshaler interfaces.
+// The zero value represents an empty filter of size 0 that uses 0 hash values.
+// Current implementations of Insert and Contains require len(f) <= 2^16 bits and k <= 16.
```

Comment:
This line refers to `f` and `k`, which aren't visible in godoc, so the comment won't make sense to the reader. It looks like this in the terminal:

```
$ go doc -all
...
type Filter struct {
	// Has unexported fields.
}
    Filter represents a Bloom filter. Filter satisfies the
    encoding.BinaryMarshaler and BinaryUnmarshaler interfaces. The zero value
    represents an empty filter of size 0 that uses 0 hash values. Current
    implementations of Insert and Contains require len(f) <= 2^16 bits and k <=
    16.
...
```

I'd suggest moving this line to `New` (adjusted appropriately for f vs b).

Also, this is minor, but there's no need to say things like "Current implementations of". It is understood that the documentation is supposed to describe the current implementation at any given time. If the implementation changes, the docs should too. I guess if you expect the requirements to be relaxed in the future you might say that, but it's hard to predict the future lol.

Another nit: I would avoid writing "2^16" to mean exponentiation because `^` is bitwise xor in Go so it might confuse the reader.